corehq/apps/data_dictionary/models.py

```diff
@@ -151,7 +151,7 @@ def clear_caches(cls, domain, case_type):
         )
         get_data_dict_props_by_case_type.clear(domain)
         get_gps_properties.clear(domain, case_type)
-        if toggles.DATA_CLEANING_CASES.enabled(domain):
+        if domain_has_privilege(domain, privileges.BULK_DATA_EDITING):
```

Comment:
should clearing the caches be conditional on whether or not a domain has the privilege? It feels like this introduces an edge case where a domain has data cleaning, has an open session, then downgrades and loses privileges on data cleaning. They do something to trigger `clear_caches` (I'm guessing release a new app build), which, because they don't have the privilege anymore, does not clear the data cleaning cache. Then, when funding is restored, they upgrade and once again have data cleaning. At that point, they'd see the stale data from the cache, right?

Reply:
it doesn't really matter, but i was following the pattern of the line below it, which also relies on a privledge